flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkContainers.java

```diff
@@ -160,6 +162,7 @@ public void stop() {
             restClusterClient.close();
         }
         this.taskManagers.forEach(GenericContainer::stop);
+        deleteJobManagerTemporaryFiles();
```

Comment:
Do we still need the cleanup hook passed to this class? AFAICT the jobmanager deletes the HA and the checkpoint data so only the image has to be deleted. Can we delete the image immediately after we have built the containers in the builder already?

Reply:
I prefer to keep it for deleting temporary directories that are created in `FlinkContainersBuilder`. JM only removes dirs and files it created inside the temp dir (`/flink/checkpoint/{jobid}/xxx`) but not the temp dir itself (`/flink/checkpoint`).

Currently the image will be deleted when JVM exits. The image will be kept only if you use `new ImageFromDockerfile(name, false)`, where the second parameter is `deleteOnExit`.
Code for building Flink image: https://github.com/apache/flink/blob/master/flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/container/FlinkImageBuilder.java#L215